Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -4246,13 +4261,64 @@ namespace ScriptCanvasEditor
             RemoveSystemTickAction(SystemTickActionFlag::CloseNextTabAction);
             CloseNextTab();
         }
+    }
+
+    void MainWindow::ClearSaveAttempt()
+    {
+        m_saveAttemptInProgress = false;
 
-        if (m_systemTickActions == 0)
+        if (!m_systemTickActions)
         {
             AZ::SystemTickBus::Handler::BusDisconnect();
         }
     }
 
+    bool MainWindow::EvaluateSaveAttempt()
+    {
+        const AZ::s64 SaveAttemptSeconds = 20;
+
+        if (m_saveAttemptInProgress)
+        {
+            auto saveDuration = AZStd::chrono::seconds(AZStd::chrono::system_clock::now() - m_saveAttemptTime).count();
+
+            if (saveDuration > SaveAttemptSeconds)
+            {
+                WarnOnFailedSaveAttempt();
+            }
+        }
+
+        return m_forceCloseInProgress;
```

Comment:
nit: There doesn't look like there is a need to return this flag (or anything at all) since the m_forceCloseInProgress is a member variable of the class and the calls to this function doesnt act on it